tests/ignite/metrics/test_ndcg.py

```diff
@@ -0,0 +1,145 @@
+import numpy as np
+import pytest
+import torch
+from sklearn.metrics import ndcg_score
+from sklearn.metrics._ranking import _dcg_sample_scores
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.recsys.ndcg import NDCG
+
+
+@pytest.fixture(params=[item for item in range(6)])
+def test_case(request):
+
+    return [
+        (torch.tensor([[3.7, 4.8, 3.9, 4.3, 4.9]]), torch.tensor([[2.9, 5.6, 3.8, 7.9, 6.2]])),
+        (
+            torch.tensor([[3.7, 3.7, 3.7, 3.7, 3.7], [3.7, 3.7, 3.7, 3.7, 3.9]]),
+            torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0], [1.0, 2.0, 3.0, 4.0, 5.0]]),
+        ),
+    ][request.param % 2]
+
+
+@pytest.mark.parametrize("k", [None, 2, 3])
+@pytest.mark.parametrize("exponential", [True, False])
+@pytest.mark.parametrize("ignore_ties_flag", [True, False])
+@pytest.mark.parametrize("ignore_ties_input", [True, False])
```

Comment:
Maybe, we can rename them for better understanding like :
```
ignore_ties_flag -> ignore_ties
ignore_ties_input -> replacement
```

On the other hand, if we have ties in the input data but we compute metrics with `ignore_ties=True`, this would lead to incorrect computation ?
Maybe, we can parametrize without this test case:
```suggestion
@pytest.mark.parametrize("ignore_ties, replacement", [(True, False), (False, True), (False, False)])
```

Reply:
In that case I just return out of the function without performing any tests. But sure, this is better.